rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -234,13 +251,18 @@ class rocalTensor {
     explicit rocalTensor(const rocalTensorInfo& tensor_info);
     int create(vx_context context);
     void update_tensor_roi(const std::vector<uint32_t>& width, const std::vector<uint32_t>& height);
+    void update_tensor_orig_roi(const std::vector<uint32_t> &width, const std::vector<uint32_t> &height);
+    void update_audio_tensor_sample_rate(const std::vector<float>& sample_rate);
     void reset_tensor_roi() { _info.reset_tensor_roi_buffers(); }
+    void swap_tensor_roi(std::shared_ptr<unsigned> &roi_ptr) { _info.swap_roi_ptr(roi_ptr); }
+    void reset_audio_sample_rate() {_info.reallocate_tensor_sample_rate_buffers();}
     // create_from_handle() no internal memory allocation is done here since
     // tensor's handle should be swapped with external buffers before usage
     int create_from_handle(vx_context context);
     int create_virtual(vx_context context, vx_graph graph);
     bool is_handle_set() { return (_vx_handle != 0); }
-    void set_dims(std::vector<size_t> dims) { _info.set_dims(dims); }
+    void set_dims(std::vector<size_t>& dims) { _info.set_dims(dims); }
+    void copy_roi(void *roi_buffer) { _info.copy_roi(roi_buffer); }
```

Comment:
How is this copy_roi different from what has been declared earlier in the file

Reply:
copy_roi does the copy whereas swap roi does the shared pointers